### PR TITLE
fix: Exotel Call Log fixes

### PR DESCRIPTION
**Pull request by nemesis189**
- When creating a call the status was always set as ringing even for missed calls, fixed to show the correct status:
   **No Answer** if no agent picks up the call
   **Canceled** if the customer hangs up before any agent picks up the call 

- Added field to display name of employee to whom the call is directed towards

### CODE REVIEW

File: erpnext/erpnext_integrations/exotel_integration.py [hunk 1]
```diff
@@ -33,10 +33,25 @@ def handle_end_call(**kwargs):
 
 @frappe.whitelist(allow_guest=True)
 def handle_missed_call(**kwargs):
-	update_call_log(kwargs, 'Missed')
+	status = ""
+	CallType = kwargs.get("CallType")
```
Comment (surajshetty3416) [inline erpnext/erpnext_integrations/exotel_integration.py]:
```suggestion
	call_type = kwargs.get("CallType")
```
according to our naming convention, we use CapitalCase for class names or for query builder instances and snake case for variables

File: erpnext/public/js/call_popup/call_popup.js [hunk 2]
```diff
@@ -140,6 +140,15 @@ class CallPopup {
 			}, {
 				'fieldtype': 'Section Break',
 				'hide_border': 1,
+			}, {
+				'fieldname': 'type_of_call',
+				'label': 'Type Of Call',
+				'fieldtype': 'Select',
+				'options': '\nFrappe Cloud Queries/Plan\nEnterprise Plans\nPartnership\nSupport\nBilling/Renewal\nOpen source / Junk',
```
Comment (surajshetty3416) [inline erpnext/public/js/call_popup/call_popup.js]: 👀

"Frappe Cloud Queries" or other options here are not really relevant for all telephony users... better to have a link field instead.

File: erpnext/public/js/call_popup/call_popup.js [hunk 3]
```diff
@@ -140,6 +140,15 @@ class CallPopup {
 			}, {
 				'fieldtype': 'Section Break',
 				'hide_border': 1,
+			}, {
+				'fieldname': 'type_of_call',
+				'label': 'Type Of Call',
```
Comment (surajshetty3416) [inline erpnext/public/js/call_popup/call_popup.js]:
```suggestion
				'label': 'Call Type',
```
simpler?

File: erpnext/telephony/doctype/call_log/call_log.json [hunk 4]
```diff
@@ -134,15 +138,34 @@
    "fieldname": "call_details_section",
    "fieldtype": "Section Break",
    "label": "Call Details"
+  },
+  {
+   "depends_on": "to",
+   "fieldname": "employee_call_directed_to",
+   "fieldtype": "Data",
+   "label": "Employee Call Directed To",
```
Comment (surajshetty3416) [inline erpnext/telephony/doctype/call_log/call_log.json]:
```suggestion
   "label": "Call Received By",
```
?

File: erpnext/telephony/doctype/call_log/call_log.py [hunk 5]
```diff
@@ -33,6 +33,16 @@ def before_insert(self):
 		if lead:
 			self.add_link(link_type='Lead', link_name=lead)
 
+		# Add Employee Name
+		if self.is_incoming_call():
+			# Taking the last 10 digits of the number
+			emp_number_reversed = (self.get("to"))[-1:-11:-1]
+			emp_number = emp_number_reversed[-1::-1]
+
+			employee = frappe.get_all("Employee", filters={"cell_number":["like","%"+emp_number+"%"]}, fields=["first_name", "middle_name", "last_name", "user_id"])
```
Comment (surajshetty3416) [inline erpnext/telephony/doctype/call_log/call_log.py]:
```suggestion
			employee = frappe.get_all("Employee", filters={
				"cell_number": ["like", "%"+emp_number+"%"]
			}, fields=["first_name", "middle_name", "last_name", "user_id"])
```

**Review by surajshetty3416 — CHANGES_REQUESTED** [review]

File: erpnext/telephony/doctype/call_log/call_log.json (hunk 4 above)
Comment (nemesis189) [reply erpnext/telephony/doctype/call_log/call_log.json]: call may not always be 'received'(picked up/ attended) by the agent so i felt 'directed to' would make more sense, anyway I'll change

File: erpnext/telephony/doctype/call_log/call_log.py [hunk 6]
```diff
@@ -33,6 +33,18 @@ def before_insert(self):
 		if lead:
 			self.add_link(link_type='Lead', link_name=lead)
 
+		# Add Employee Name
+		if self.is_incoming_call():
+			# Taking the last 10 digits of the number
+			emp_number_reversed = (self.get("to"))[-1:-11:-1]
+			emp_number = emp_number_reversed[-1::-1]
```
Comment (surajshetty3416) [inline erpnext/telephony/doctype/call_log/call_log.py]: use `strip_number` instead? Also, we avoid short-forms for variable names... let's keep it `employee_number`.

File: erpnext/telephony/doctype/call_log/call_log.py [hunk 7]
```diff
@@ -33,6 +33,18 @@ def before_insert(self):
...
+			employee = frappe.get_all("Employee", filters={
+				"cell_number": ["like", "%"+emp_number+"%"]
+			}, fields=["first_name", "middle_name", "last_name", "user_id"])
```
Comment (surajshetty3416) [inline erpnext/telephony/doctype/call_log/call_log.py]: you can extend `get_employees_with_number` method to make this work.

File: erpnext/telephony/doctype/call_log/call_log.py [hunk 8]
```diff
@@ -33,6 +33,18 @@ def before_insert(self):
...
+
+			self.call_received_by = get_employee_name(employee[0])
```
Comment (surajshetty3416) [inline erpnext/telephony/doctype/call_log/call_log.py]: Let's keep `call_rececived_by` a link field. We can also, enable the [title for link](https://frappeframework.com/docs/v13/user/en/basics/doctypes/form_&_view_settings#title-field) for Employee doctype.

**Review by surajshetty3416 — CHANGES_REQUESTED** [review]

**Comment by surajshetty3416** [comment]
@nemesis189 let's also introduce basic tests for this integration.

File: erpnext/telephony/doctype/call_log/call_log.py (hunk 6 above)
Comment (nemesis189) [reply erpnext/telephony/doctype/call_log/call_log.py]: Done

File: erpnext/telephony/doctype/call_log/call_log.py (hunk 7 above)
Comment (nemesis189) [reply erpnext/telephony/doctype/call_log/call_log.py]: tweaked and used the method as suggested

**Review by surajshetty3416 — DISMISSED** [review]

**Comment by surajshetty3416** [comment]
@Mergifyio backport version-13-hotfix

**Pull request merged**